spec/controllers/infra_networking_controller_spec.rb

```diff
@@ -1,27 +1,121 @@
 describe InfraNetworkingController do
-  let(:zone) { FactoryGirl.build(:zone) }
-  let!(:server) { EvmSpecHelper.local_miq_server(:zone => zone) }
-
+  let!(:switch) { FactoryGirl.create(:switch, :name => 'test_switch1', :shared => 'true') }
+  let(:host) { FactoryGirl.create(:host, :name => 'test_host1') }
+  let(:ems_vmware) { FactoryGirl.create(:ems_vmware, :name => 'test_vmware') }
+  let(:cluster) { FactoryGirl.create(:cluster, :name => 'test_cluster') }
+  let(:user) { FactoryGirl.create(:user) }
+  let(:classification) { FactoryGirl.create(:classification, :name => "department", :description => "Department") }
+  let(:tag1) { FactoryGirl.create(:classification_tag, :name   => "tag1", :parent => classification) }
+  let(:tag2) { FactoryGirl.create(:classification_tag, :name   => "tag2", :parent => classification) }
   before { stub_user(:features => :all) }
 
-  describe '#tree_select' do
+  describe 'render_views' do
     render_views
 
-    context 'switch' do
-      let(:ems) { FactoryGirl.create(:ems_vmware) }
-      let(:cluster) { FactoryGirl.create(:ems_cluster, :ems_id => ems.id) }
-      let(:host) { FactoryGirl.create(:host_vmware, :ems_id => ems.id, :ems_cluster => cluster) }
-      let(:switch) do
-        sw = FactoryGirl.create(:switch_vmware, :ems_id => ems.id)
-        FactoryGirl.create(:host_switch, :host => host, :switch => sw)
-        sw
+    describe '#explorer' do
+      before do
+        session[:settings] = {:views => {}, :perpage => {:list => 5}}
+        EvmSpecHelper.create_guid_miq_server_zone
+      end
+
+      it 'can render the explorer' do
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+        get :explorer
+        expect(response.status).to eq(200)
+        expect(response.body).to_not be_empty
+      end
+
+      it 'shows a switch in the list' do
+        session[:sb] = {:active_accord => :infra_networking_accord}
+        seed_session_trees('switch', :infra_networking_tree, 'root')
+
+        get :explorer
+        expect(response.body).to include("modelName: 'Switch'")
+        expect(response.body).to include("activeTree: 'infra_networking_tree'")
+        expect(response.body).to include("gtlType: 'list'")
+        expect(response.body).to include("isExplorer: 'true' === 'true' ? true : false")
+        expect(response.body).to include("showUrl: '/infra_networking/x_show/'")
+      end
+    end
+
+    describe "#tree_select" do
+      [
+        ['All Distributed Switches', 'infra_networking_tree'],
```

Comment:
If you write `:infra_networking_tree` here, you don't need the `to_sym` lower :wink: